src/main/java/hudson/plugins/openid/OpenIdSession.java

```diff
@@ -87,7 +89,7 @@ public HttpResponse doCommenceLogin() throws IOException, OpenIDException {
         String url = authReq.getDestinationUrl(true);
 
         // remember this in the session
-        Stapler.getCurrentRequest().getSession().setAttribute(SESSION_NAME,this);
+        Stapler.getCurrentRequest().getSession().setAttribute(SESSION_NAME,this.toString());
```

Comment:
I don't see how `setAttribute(SESSION_NAME,this)` could have worked anyway (across restarts) since `OpenIdSession` (and anonymous impls of in this codebase) does not appear to be `Serializable`.